diff --git a/CMakeLists.txt b/CMakeLists.txt
index e70ded596d7..d947e6c6d33 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -684,23 +684,16 @@ endif ()
 install(EXPORT SUMOConfig DESTINATION ${DATA_PATH}cmake NAMESPACE SUMO::)
 
 set(SUMO_TOOLS_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tools")
-if (NOT ${PYTHON_SETUPTOOLS_MISSING})
-    if (${PYTHON_BUILD_MISSING})
-        install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} build_config/setup-sumolib.py clean --all install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} --optimize=1 WORKING_DIRECTORY ${SUMO_TOOLS_DIR})"
-                COMPONENT pysumolib)
-        install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} build_config/setup-traci.py clean --all install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} --optimize=1 WORKING_DIRECTORY ${SUMO_TOOLS_DIR})"
-                COMPONENT pytraci)
-    else ()
-        if (NOT VERBOSE_SUB)
-            set(PYTHON_BUILD_OPTS "-C--quiet")
-        endif ()
-        install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-sumolib.py ./setup.py WORKING_DIRECTORY ${SUMO_TOOLS_DIR} COMMAND ${PYTHON_EXECUTABLE} -m build ${PYTHON_BUILD_OPTS} --wheel . -o ${CMAKE_BINARY_DIR})"
-                COMPONENT pysumolib)
-        install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-traci.py ./setup.py WORKING_DIRECTORY ${SUMO_TOOLS_DIR} COMMAND ${PYTHON_EXECUTABLE} -m build ${PYTHON_BUILD_OPTS} --wheel . -o ${CMAKE_BINARY_DIR})"
-                COMPONENT pytraci)
-        install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} -m pip install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} -f ${CMAKE_BINARY_DIR} --no-index traci)"
-                COMPONENT pytraci)
+if (NOT ${PYTHON_SETUPTOOLS_MISSING} AND NOT ${PYTHON_BUILD_MISSING})
+    if (NOT VERBOSE_SUB)
+        set(PYTHON_BUILD_OPTS "-C--quiet")
     endif ()
+    install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-sumolib.py ./setup.py WORKING_DIRECTORY ${SUMO_TOOLS_DIR} COMMAND ${PYTHON_EXECUTABLE} -m build ${PYTHON_BUILD_OPTS} --wheel . -o ${CMAKE_BINARY_DIR})"
+            COMPONENT pysumolib)
+    install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-traci.py ./setup.py WORKING_DIRECTORY ${SUMO_TOOLS_DIR} COMMAND ${PYTHON_EXECUTABLE} -m build ${PYTHON_BUILD_OPTS} --wheel . -o ${CMAKE_BINARY_DIR})"
+            COMPONENT pytraci)
+    install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} -m pip install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} -f ${CMAKE_BINARY_DIR} --no-index traci)"
+            COMPONENT pytraci)
 endif ()
 
 if (SUMO_LIBRARIES AND WIN32)
diff --git a/src/libsumo/CMakeLists.txt b/src/libsumo/CMakeLists.txt
index 5e8d00bbac4..23f6d580da5 100644
--- a/src/libsumo/CMakeLists.txt
+++ b/src/libsumo/CMakeLists.txt
@@ -212,14 +212,9 @@ if (SWIG_FOUND)
             swig_link_libraries(libsumo ${PYTHON_LIBRARIES})
         endif()
         swig_link_libraries(libsumo ${sumo_link_libraries})
-        if (NOT ${PYTHON_SETUPTOOLS_MISSING})
-            if (${PYTHON_BUILD_MISSING})
-                install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SUMO_TOOLS_DIR}/build_config/setup-libsumo.py clean --all install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} --optimize=1)"
-                        COMPONENT pylibsumo)
-            else ()
-                install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-libsumo.py ./setup.py COMMAND ${PYTHON_EXECUTABLE} -m pip install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} -f ${CMAKE_BINARY_DIR} . WORKING_DIRECTORY ${SUMO_TOOLS_DIR})"
-                        COMPONENT pylibsumo)
-            endif()
+        if (NOT ${PYTHON_SETUPTOOLS_MISSING} AND NOT ${PYTHON_BUILD_MISSING})
+            install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-libsumo.py ./setup.py COMMAND ${PYTHON_EXECUTABLE} -m pip install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} -f ${CMAKE_BINARY_DIR} . WORKING_DIRECTORY ${SUMO_TOOLS_DIR})"
+                    COMPONENT pylibsumo)
         endif()
     endif()
 
diff --git a/src/libtraci/CMakeLists.txt b/src/libtraci/CMakeLists.txt
index f4841467f76..01a5905e5a5 100644
--- a/src/libtraci/CMakeLists.txt
+++ b/src/libtraci/CMakeLists.txt
@@ -165,14 +165,9 @@ if (SWIG_FOUND)
         if (WIN32)
             swig_link_libraries(libtraci ${PYTHON_LIBRARIES})
         endif()
-        if (NOT ${PYTHON_SETUPTOOLS_MISSING})
-            if (${PYTHON_BUILD_MISSING})
-                install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ${SUMO_TOOLS_DIR}/build_config/setup-libtraci.py clean --all install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} --optimize=1)"
-                        COMPONENT pylibtraci)
-            else ()
-                install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-libtraci.py ./setup.py COMMAND ${PYTHON_EXECUTABLE} -m pip install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} -f ${CMAKE_BINARY_DIR} . WORKING_DIRECTORY ${SUMO_TOOLS_DIR})"
-                        COMPONENT pylibtraci)
-            endif()
+        if (NOT ${PYTHON_SETUPTOOLS_MISSING} AND NOT ${PYTHON_BUILD_MISSING})
+            install(CODE "execute_process(COMMAND ${PYTHON_EXECUTABLE} ./build_config/version.py ./build_config/setup-libtraci.py ./setup.py COMMAND ${PYTHON_EXECUTABLE} -m pip install --root=\$ENV{DESTDIR}/ --prefix=${CMAKE_INSTALL_PREFIX} -f ${CMAKE_BINARY_DIR} . WORKING_DIRECTORY ${SUMO_TOOLS_DIR})"
+                    COMPONENT pylibtraci)
         endif()
     endif()
 
