diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fc734a6499e..e1880bb4897 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -68,7 +68,6 @@ endif ()
 add_executable(sumo sumo_main.cpp)
 set_target_properties(sumo PROPERTIES OUTPUT_NAME sumo${BINARY_SUFFIX})
 set_target_properties(sumo PROPERTIES OUTPUT_NAME_DEBUG sumo${BINARY_SUFFIX}D)
-set_target_properties(sumo PROPERTIES OUTPUT_NAME_RELWITHDEBINFO sumo${BINARY_SUFFIX}R)
 target_link_libraries(sumo ${sumolibs} libsumostatic ${TCMALLOC_LIBRARY})
 add_dependencies(sumo generate-version-h install_dll)
 
@@ -76,7 +75,6 @@ if (FOX_FOUND)
     add_executable(sumo-gui guisim_main.cpp sumo-gui.rc)
     set_target_properties(sumo-gui PROPERTIES OUTPUT_NAME sumo-gui${BINARY_SUFFIX})
     set_target_properties(sumo-gui PROPERTIES OUTPUT_NAME_DEBUG sumo-gui${BINARY_SUFFIX}D)
-    set_target_properties(sumo-gui PROPERTIES OUTPUT_NAME_RELWITHDEBINFO sumo-gui${BINARY_SUFFIX}R)
     target_link_libraries(sumo-gui ${sumoguilibs} libsumoguistatic ${TCMALLOC_LIBRARY})
     if (MSVC)
         if (CONSOLE_RELEASE)
